src/platform/plugins/shared/unified_doc_viewer/public/components/doc_viewer_traces_overview/sub_components/trace_summary.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { TRANSACTION_NAME_FIELD } from '@kbn/discover-utils';
+import { EuiHorizontalRule } from '@elastic/eui';
+import React, { useState, useEffect } from 'react';
+import { useTransactionContext } from '../hooks/use_transaction';
+import { FieldWithActions } from './field_with_actions/field_with_actions';
+import { FieldConfiguration } from '../resources/get_field_configuration';
+
+export interface TraceSummaryProps {
+  fieldId: string;
+  fieldConfiguration: FieldConfiguration;
+}
+
+export function TraceSummary({ fieldConfiguration, fieldId }: TraceSummaryProps) {
+  const { transaction, loading } = useTransactionContext();
+  const [fieldValue, setFieldValue] = useState(fieldConfiguration.value);
+  const isTransactionNameField = fieldId === TRANSACTION_NAME_FIELD;
+  const displayLoading = isTransactionNameField && !fieldValue;
```

Comment:
Shouldn't `loading` also be counted here?
```suggestion
  const displayLoading = isTransactionNameField && !fieldValue && loading;
```

Reply:
Not exactly.

The name `displayLoading` was confusing because it didn't fully reflect its purpose, so I renamed it to `isTransactionNameFieldWithoutValue`.

Now, when using it in the template, the goal is to only show the loader if it's the transaction name field and it's loading.

I’m not entirely comfortable with having such a specific exception in a generic component, and I left a comment in the file about it. However, for the sake of simplicity, I decided to leave it as is for now.

https://github.com/elastic/kibana/pull/210612/commits/62479f64bf6c8ad420bddadaf465cdff3d106387